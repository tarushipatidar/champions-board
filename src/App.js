import React from 'react';
import Dashboard from './components/Dashboard';
import WatchListPage from './components/WatchListPage'
import { BrowserRouter, Routes, Route } from "react-router-dom";
import HeaderFooter from './components/HeaderFooter';
import './stylesheets/index.js';

function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<HeaderFooter />} >
          <Route index element={<Dashboard />} />
          <Route path="/watchedchampions" element={<WatchListPage />} />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default App;
