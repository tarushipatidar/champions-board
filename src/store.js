import { configureStore } from "@reduxjs/toolkit";
import updateWatchList from "./reducers/addRemoveWatchItem";
import updateCurrentUser from "./reducers/logInOutUser"

const store = configureStore({
  reducer: {
    updateWatchList: updateWatchList,
    updateCurrentUser: updateCurrentUser,
  }
})

export default store;
