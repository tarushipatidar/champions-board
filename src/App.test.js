import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import store from "./store"
import { GoogleOAuthProvider } from '@react-oauth/google';

describe('With React Testing Library', () => {
  let root = document.createElement('div');
  let clientId = '468227137198-dutfsgqd0rnv383oe38ca1thgi6r1mbg.apps.googleusercontent.com';
  document.body.appendChild(root);

  it('Shows "Champions", "LogIn With Google" and "Champions Board"', () => {
      const { getByText } = render(
        <GoogleOAuthProvider clientId={ clientId }>
          <Provider store={ store }>
              <App />
          </Provider>
        </GoogleOAuthProvider>
      );

      expect(getByText('Champions')).not.toBeNull();
      expect(getByText('LogIn With Google')).not.toBeNull();
      expect(getByText('Champions Board')).not.toBeNull();
      expect(getByText).toMatchSnapshot();
  });
});
