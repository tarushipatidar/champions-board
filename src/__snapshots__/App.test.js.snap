// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Shows "Champions", "LogIn With Google" and "Champions Board" 1`] = `[Function]`;
